plugins/experimental/ts_lua/ts_lua_transform.c

```diff
@@ -231,13 +261,19 @@ ts_lua_transform_handler(TSCont contp, ts_lua_http_transform_ctx *transform_ctx,
   if (toread > input_avail) { // upstream not finished.
     if (eos) {
       TSVIONBytesSet(transform_ctx->output.vio, transform_ctx->total);
-      TSContCall(TSVIOContGet(input_vio), TS_EVENT_VCONN_EOS, input_vio);
+      if (empty_input == 0) {
+        TSContCall(TSVIOContGet(input_vio), TS_EVENT_VCONN_EOS, input_vio);
```

Comment:
You can't `TSContCall` here for `empty_input == 1` because there's no `input_vio`? But that seems to be set early in the function. It might be reasonable to send `TS_EVENT_VCONN_EOS` for empty input.

Reply:
I find that it is not needed in this case.